server/app/services/barman-service.js

```diff
@@ -144,6 +151,12 @@ async function updateBarmanById(barmanId, updatedBarman, _embedded) {
 
   if (!currentBarman) throw createUserError('UnknownBarman', 'This Barman does not exist');
 
+  if ((updatedBarman.leaveAt
+    && new Date(currentBarman.createdAt) > new Date(currentBarman.leaveAt))
```

Comment:
createdAt and leaveAt should already be a date no?

Reply:
console.log(typeof(currentBarman.createdAt)) => object.
So I need to use Date constructor